src/jllvm/materialization/JIT2InterpreterLayer.cpp

```diff
@@ -38,77 +36,62 @@ void jllvm::JIT2InterpreterLayer::emit(std::unique_ptr<llvm::orc::Materializatio
-            // the GC mask as well.
-            std::size_t localVariableIndex = 0;
-            if (!method->isStatic())
-            {
-                // Store the 'this' argument.
-                llvm::Value* gep = builder.CreateConstGEP1_32(builder.getInt64Ty(), localVariables, localVariableIndex);
-                builder.CreateStore(function->getArg(0), gep);
-                llvm::Value* maskGep =
-                    builder.CreateConstGEP1_32(builder.getInt64Ty(), localVariablesGCMask, localVariableIndex / 64);
-                llvm::Value* value = builder.CreateLoad(builder.getInt64Ty(), maskGep);
-                value = builder.CreateOr(value, builder.getInt64(1 << (localVariableIndex % 64)));
-                builder.CreateStore(value, maskGep);
-                localVariableIndex++;
-            }
-
-            for (auto&& [argument, paramType] : llvm::zip_equal(llvm::drop_begin(function->args(), localVariableIndex),
-                                                                method->getType().parameters()))
-            {
-                llvm::Value* gep = builder.CreateConstGEP1_32(builder.getInt64Ty(), localVariables, localVariableIndex);
-                builder.CreateStore(&argument, gep);
-                if (paramType.isReference())
-                {
-                    llvm::Value* maskGep =
-                        builder.CreateConstGEP1_32(builder.getInt64Ty(), localVariablesGCMask, localVariableIndex / 64);
-                    llvm::Value* value = builder.CreateLoad(builder.getInt64Ty(), maskGep);
-                    value = builder.CreateOr(value, builder.getInt64(1 << (localVariableIndex % 64)));
-                    builder.CreateStore(value, maskGep);
-                }
-
-                localVariableIndex++;
-                if (paramType.isWide())
-                {
-                    localVariableIndex++;
-                }
-            }
-        });
-    if (returnValue)
+            argumentArrayCount++;
+        }
+    }
+
+    llvm::Value* argumentArray = builder.CreateAlloca(llvm::ArrayType::get(builder.getInt64Ty(), argumentArrayCount));
+    // Zero out argument array for any unassigned bytes.
+    builder.CreateMemSet(argumentArray, builder.getInt8(0), argumentArrayCount * sizeof(std::uint64_t),
+                         llvm::MaybeAlign());
+    std::size_t index = 0;
+    for (llvm::Argument& arg : function->args())
     {
-        builder.CreateRet(returnValue);
+        llvm::Value* gep = builder.CreateConstGEP1_32(builder.getInt64Ty(), argumentArray, index);
+        builder.CreateStore(&arg, gep);
+        index++;
+        if (arg.getType()->isDoubleTy() || arg.getType()->isIntegerTy(64))
+        {
+            index++;
+        }
     }
-    else
+
+    llvm::Value* value = builder.CreateCall(module->getOrInsertFunction("Interpreter Entry", builder.getInt64Ty(),
```

Comment:
To keep the convention:
```suggestion
    llvm::Value* value = builder.CreateCall(module->getOrInsertFunction("jllvm_interpreter_entry", builder.getInt64Ty(),
```